src/olympia/migrations/1127-db-schema-fixes2.sql

```diff
@@ -0,0 +1,129 @@
+
+ALTER TABLE `akismet_reports`
+    MODIFY `rating_instance_id` int(10) unsigned DEFAULT NULL,
+    MODIFY `addon_instance_id` int(10) unsigned DEFAULT NULL,
+    MODIFY `collection_instance_id` int(10) unsigned DEFAULT NULL,
+    MODIFY `user_id` int(11) DEFAULT NULL,
+    DROP FOREIGN KEY `akismet_reports_addon_instance_id_fk_addons_id`,
+    DROP FOREIGN KEY `akismet_reports_collection_instance_id_fk_collections_id`,
+    DROP FOREIGN KEY `akismet_reports_upload_instance_id_fk_file_uploads_id`,
+    DROP FOREIGN KEY `akismet_reports_user_id_fk_users_id`;
+ALTER TABLE `akismet_reports`
+    ADD CONSTRAINT `akismet_reports_addon_instance_id_03f471af_fk_addons_id` FOREIGN KEY (`addon_instance_id`) REFERENCES `addons` (`id`),
+    ADD CONSTRAINT `akismet_reports_collection_instance__2c06adf6_fk_collectio` FOREIGN KEY (`collection_instance_id`) REFERENCES `collections` (`id`),
+    ADD CONSTRAINT `akismet_reports_upload_instance_id_c4530dc1_fk_file_uploads_id` FOREIGN KEY (`upload_instance_id`) REFERENCES `file_uploads` (`id`),
+    ADD CONSTRAINT `akismet_reports_user_id_97ce80b4_fk_users_id` FOREIGN KEY (`user_id`) REFERENCES `users` (`id`);
+
+ALTER TABLE `api_key`
+    MODIFY `id` int(10) unsigned NOT NULL,
+    MODIFY `created` datetime(6) NOT NULL,
+    MODIFY `modified` datetime(6) NOT NULL,
+    MODIFY `type` int(10) unsigned NOT NULL,
+    DROP FOREIGN KEY `api_key_user_id`;
+ALTER TABLE `api_key`
+    ADD CONSTRAINT `api_key_user_id_2b8305f7_fk_users_id` FOREIGN KEY (`user_id`) REFERENCES `users` (`id`);
+
+ALTER TABLE `applications_versions`
+    MODIFY `application_id` int(10) unsigned NOT NULL,
+    MODIFY `version_id` int(10) unsigned NOT NULL,
+    MODIFY `min` int(10) unsigned NOT NULL,
+    MODIFY `max` int(10) unsigned NOT NULL,
+    DROP FOREIGN KEY `applications_versions_ibfk_4`,  /* `version_id` */
+    DROP FOREIGN KEY `applications_versions_ibfk_5`,  /* `min` */
+    DROP FOREIGN KEY `applications_versions_ibfk_6`;  /* `max` */
+ALTER TABLE `applications_versions`
+    ADD CONSTRAINT `applications_versions_version_id_9bf048e6_fk_versions_id` FOREIGN KEY (`version_id`) REFERENCES `versions` (`id`),
+    ADD CONSTRAINT `applications_versions_min_1c31b27c_fk_appversions_id` FOREIGN KEY (`min`) REFERENCES `appversions` (`id`),
+    ADD CONSTRAINT `applications_versions_max_6e57db5a_fk_appversions_id` FOREIGN KEY (`max`) REFERENCES `appversions` (`id`);
+
+ALTER TABLE `appsupport`
+    MODIFY `id` int(10) unsigned NOT NULL,
+    MODIFY `created` datetime(6) NOT NULL,
+    MODIFY `modified` datetime(6) NOT NULL,
+    MODIFY `addon_id` int(10) unsigned NOT NULL,
+    MODIFY `app_id` int(10) unsigned NOT NULL,
+    MODIFY `min` bigint(20) DEFAULT NULL,
+    MODIFY `max` bigint(20) DEFAULT NULL,
+    DROP KEY `app_id_refs_id_481ce338`;  /* (`app_id`)  note: there's the key, but no existing fk constraint */
```

Comment:
Having an index on this is probably useful though. We do filter on it. I'd say keep ?

Reply:
https://github.com/mozilla/addons-server/pull/12229/commits/adfe7c4dcd14bf6c4ba664dd55d86c527217ad98